TodoApp/TodoApp.Api.Tests/ItemsControllerTests.cs

```diff
@@ -0,0 +1,128 @@
+    {
+        private ItemsController _itemsController;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _itemsController = new ItemsController
+            {
+                Request = new HttpRequestMessage(),
+                Configuration = new HttpConfiguration()
+            };
+        }
+
+        [Test]
+        public async Task GetAllItemsAsync_ReturnsAllItems()
+        {
+            var expected = ItemsController.Items;
+
+            var response = await _itemsController.GetAllItemsAsync();
+
+            Assert.That(response, Is.EquivalentTo(expected));
+        }
+
+        [Test]
+        public async Task GetItemByIdAsync_ReturnsOkWithRequiredItem()
+        {
+            const string id = "1";
+            var expected = ItemsController.Items[0];
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.GetItemByIdAsync(id));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.IsEqual(expected), Is.True);
+            });
+        }
+
+        [Test]
+        public async Task PutItemAsync_ReturnsNoContent()
+        {
+            const string id = "1";
+            var expected = ItemsController.Items[2];
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.PutItemAsync(id, expected));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.IsEqual(expected), Is.True);
+            });
+        }
+
+        [Test]
+        public async Task DeleteItemAsync_ReturnsOkWithDeletedItem()
+        {
+            const string id = "3";
+            var expected = ItemsController.Items[3];
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.DeleteItemAsync(id));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.IsEqual(expected), Is.True);
+            });
+        }
+
+        [Test]
+        public async Task PostItemAsync_ReturnsCreatedWithLinkToItem()
+        {
+            _itemsController.Request = new HttpRequestMessage
+            {
+                RequestUri = new Uri("http://localhost/api/v1/items")
+            };
+            _itemsController.Configuration.Routes.MapHttpRoute(
```

Comment:
It is nice you think about this right now. It is too complex right now and we will rework this in the future tasks 👍